jupyter/scripts/incremental_learning/experiment_driver.py

```diff
@@ -116,69 +116,98 @@ def compute_classification_metrics(y_true, baseline_model_predictions, trained_m
     return scores
 
 
-@ex.main
-def my_main(_run, dataset_name, dataset_size, test_size, seed):
-    results = {}
+@ex.capture
+def read_dataset(_run, dataset_name):
     download_successful = download_dataset(dataset_name)
     if download_successful == False:
         _run.run_logger.error("Data is not available")
         exit(1)
-    D1 = pd.read_csv(datasets_dir / '{}.csv'.format(dataset_name))
-    D1.drop_duplicates(inplace=True)
-    # sample if size is specified in the config
-    if dataset_size and dataset_size < D1.shape[0]:
-        D1 = D1.sample(dataset_size)
+    dataset = pd.read_csv(datasets_dir / '{}.csv'.format(dataset_name))
+    dataset.drop_duplicates(inplace=True)
+    return dataset
+
+
+@ex.capture
+def transform_dataset(dataset: pd.DataFrame, transform_name: str, transform_parameters, _seed: int) -> pd.DataFrame:
+    if transform_name == 'resample_metric_features':
+        transformed_dataset = resample_metric_features(data_frame=dataset,
+                                                       seed=_seed, 
+                                                       fraction=transform_parameters['fraction'], 
+                                                       magnitude=transform_parameters['magnitude'],
+                                                       features=transform_parameters['features'])
+        return transformed_dataset
+    # TODO extend to other transforms
+    else:
+        raise NotImplementedError("This dataset transform is not integrated yet.")
+
+
+@ex.main
+def my_main(_run, dataset_name, test_fraction):
+    results = {}
+
+    original_dataset = read_dataset()
+    train_dataset, test1_dataset = train_test_split(original_dataset, test_size=test_fraction)
+    update_dataset = transform_dataset(dataset=train_dataset)
+    test2_dataset = transform_dataset(dataset=test1_dataset)
+
+    baseline_dataset = pd.concat([train_dataset, update_dataset])
+    test_dataset = pd.concat([test1_dataset, test2_dataset])
+    
     _run.run_logger.info("Baseline training started")
-    job1 = train(dataset_name, D1, verbose=False, run=_run)
-    elapsed_time = job1.wait_to_complete(clean=False)
+    baseline = train(dataset_name, baseline_dataset, verbose=False, run=_run)
```

Comment:
nit: for consistency
```suggestion
    baseline_model = train(dataset_name, baseline_dataset, verbose=False, run=_run)
```